libmachine/mcndockerclient/docker_client.go

```diff
@@ -19,7 +21,13 @@ func DockerClient(dockerHost DockerHost) (*dockerclient.DockerClient, error) {
 		return nil, fmt.Errorf("Unable to read TLS config: %s", err)
 	}
 
-	return dockerclient.NewDockerClient(url, tlsConfig)
+	client, err := dockerclient.NewDockerClient(url, tlsConfig)
+	if err != nil {
+		transport := (client.HTTPClient.Transport).(*http.Transport)
+		transport.Proxy = http.ProxyFromEnvironment
```

Comment:
This is the part I'm uncertain about - whether it is sane to update the proxy setting in this way.